src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerListener.cs

```diff
@@ -148,5 +177,235 @@ private void InitializeBuilder()
+                }
+
+                workPerPartition = await this._workEstimator.GetEstimatedRemainingWorkPerPartitionAsync();
+
+                if (workPerPartition != null)
+                {
+                    partitionCount = workPerPartition.Count;
+                    remainingWork = workPerPartition.Sum(item => item.RemainingWork);
+                }
+            }
+            catch (Microsoft.Azure.Documents.DocumentClientException e)
+            {
+                if (!TryHandleDocumentClientException(e))
+                {
+                    _logger.LogWarning("Unable to handle Microsoft.Azure.Documents.DocumentClientException: {0}", e.Message);
+                }
+            }
+            catch (NullReferenceException e)
+            {
+                // This shouldn't be hit because we're not using ChangeFeedHost, and WorkEstimator throws DocumentClientException
+                // for missing collections.
+                _logger.LogWarning(e.Message);
+            }
+            catch (InvalidOperationException e)
+            {
+                if (!TryHandleDocumentClientException(e))
+                {
+                    throw;
+                }
+            }
+            catch (System.Net.Http.HttpRequestException e)
+            {
+                string errormsg;
+
+                var webException = e.InnerException as WebException;
+                if (webException != null &&
+                    webException.Status == WebExceptionStatus.ProtocolError)
+                {
+                    string statusCode = ((HttpWebResponse)webException.Response).StatusCode.ToString();
+                    string statusDesc = ((HttpWebResponse)webException.Response).StatusDescription;
+                    errormsg = string.Format("CosmosDBtrigger status {0}: {1}.", statusCode, statusDesc);
+                }
+                else if (webException != null &&
+                    webException.Status == WebExceptionStatus.NameResolutionFailure)
+                {
+                    errormsg = string.Format("CosmosDBtrigger Exception message: {0}.", webException.Message);
+                }
+                else
+                {
+                    errormsg = e.ToString();
+                }
+
+                _logger.LogWarning(errormsg);
+            }
+
+            return new CosmosDBTriggerMetrics
+            {
+                TimeStamp = DateTime.UtcNow,
+                PartitionCount = partitionCount,
+                RemainingWork = remainingWork
+            };
+        }
+
+        ScaleStatus IScaleMonitor.GetScaleStatus(ScaleStatusContext context)
+        {
+            ScaleStatus status = new ScaleStatus
+            {
+                Vote = ScaleVote.None
+            };
+
+            const int NumberOfSamplesToConsider = 5;
+            List<CosmosDBTriggerMetrics> metrics = (context as ScaleStatusContext<CosmosDBTriggerMetrics>)?.Metrics?.ToList();
```

Comment:
Need to update this casting code - see my latest QueueListener for an example: https://github.com/Azure/azure-webjobs-sdk/blob/8a459a6cff394e4f1d0a5418e731024e7674256e/src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/Listeners/QueueListener.cs#L453

If you update your tests to test both the generic and non-generic interface methods, you'll find this code will fail as is